pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -552,13 +552,12 @@ def __resolve_ckpt_dir(self, trainer):
                 trainer.logger.version
                 if isinstance(trainer.logger.version, str) else f"version_{trainer.logger.version}"
             )
-
-            version, name = trainer.training_type_plugin.broadcast((version, trainer.logger.name))
-
-            ckpt_path = os.path.join(save_dir, str(name), version, "checkpoints")
+            ckpt_path = os.path.join(save_dir, str(trainer.logger.name), version, "checkpoints")
         else:
             ckpt_path = os.path.join(trainer.weights_save_path, "checkpoints")
 
+        ckpt_path = trainer.training_type_plugin.broadcast(ckpt_path)
```

Comment:
@SeanNaren Might this be an issue for DeepSpeed ?

Reply:
@tchaton @SeanNaren I'd assume we want to save all the checkpoint shards to the same directory right?

Reply:
We didn't enforce this, but I guess it is fine.